dmoj/judge.py

```diff
@@ -194,11 +196,25 @@ def _ipc_compile_message(self, _report, compile_message: str) -> None:
         self.packet_manager.compile_message_packet(compile_message)
 
     def _ipc_grading_begin(self, _report, is_pretested: bool) -> None:
+        self.in_pretests = False
+        self.pretest_batch = 0
```

Comment:
Can we use a single set of batch/case variables, and reset them when we end pretests? That way we don't have dangling state after pretests finish.